sktime/dists_kernels/scipy_dist.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+"""
+Interface module to scipy.spatial's pairwise distance function cdist
+    exposes parameters as scikit-learn hyper-parameters
+"""
+
+__author__ = ["fkiraly"]
+
+import pandas as pd
+
+from scipy.spatial.distance import cdist
+
+from sktime.dists_kernels._base import BaseTrafoPw
+
+
+class ScipyDist(BaseTrafoPw):
```

Comment:
```suggestion
class PairwiseScipyDist(BaseTrafoPw):
```
Similar to above, I think Pairwise should be part of class names

Reply:
a distance is always pairwise, I find this redundant

Reply:
maybe the word "pairwise" is not good - what this is supposed to mean is "of pairs"

Reply:
Pairwise is standard term for doing stuff to each pair of two sets/lists, I think.

Reply:
E.g. see use of pairwise in these contexts:
https://mathworld.wolfram.com/Pairwise.html
https://en.wikipedia.org/wiki/Pairwise_independence
https://en.wikipedia.org/wiki/Disjoint_sets

Reply:
yes, but my problem is not that I think it's inaccurate, just that it's redundant.

Reply:
Pairwise tells us we are operating on two list-like objects, and calculating for every pair between those two lists.

Without pairwise, I think `ScipyDist` best describes something like the euclidean distance function, which does a single distance calculation.

Reply:
>Pairwise tells us we are operating on two list-like objects, and calculating for every pair between those two lists.

by the same logic you would call a `sklearn` transformer not a transformer but an element-wise transformer, because it calculates the transformation individually for each instance in the data frame or array.

Reply:
> by the same logic you would call a `sklearn`an element-wise transformer

I see the logic, but the element-wise transformer is well-established as the default or base concept.  The question we should be asking ourselves is: if a user saw the class `ScipyDist` for the first time, would they predict it does calculations elementwise or pairwise? If you find *anybody* whose first thought is that it does pairwise calculations, I will be very surprised.

Reply:
Interestingly, it is curious how we are on opposites sides of the same argument.

* I prefer `BasePairwiseTransformerTabular` and `BasePairwiseTransformerPanel`, whereas you prefer having `BasePairwiseTransformer`
* In the hypothetical scenario where we created both pairwise and element-wise wrappers for scipy dist, I would prefer `PairwiseScipyDist` and `ScipyDist` whereas you (appear to) prefer `PairwiseScipyDist` and `ElementwiseScipyDist`

I am comfortable being on different sides, because I think the element-wise calculations are a well-established norm and base, whereas I don't think any norms are established for BasePairwiseTransformers.